### PR TITLE
Fix lost results on query editor save

### PR DESCRIPTION
**Pull request by kburtram**
This PR fixes #https://github.com/microsoft/azuredatastudio/issues/21038.  The `replaceEditor` functionality is different post-merge and overwrites the input state when resolving editors.  Previous implementation didn't resolveEditor during the replace operation.  Also, the return value from `createFileQueryEditorInput` was now an untyped editor input rather than a query editor input.

### CODE REVIEW

File: src/sql/workbench/contrib/query/browser/queryEditor.ts [hunk 1]
```diff
@@ -312,10 +312,10 @@ export class QueryEditor extends EditorPane {
 
 		// TODO: Allow query provider to provide the language mode.
 		if (this.input instanceof UntitledQueryEditorInput) {
-			if ((providerId === 'KUSTO') || this.languageService.getExtensions('Kusto').indexOf(fileExtension) > -1) {
+			if ((providerId === 'KUSTO') || this.languageService.getExtensions('kusto').indexOf(fileExtension) > -1) {
```
Comment (kburtram) [inline src/sql/workbench/contrib/query/browser/queryEditor.ts:315]: unrelated change, but now getExtensions expects language ID instead of language name.  Likely the provider test would be true in any case, but there could be some new corner case when this test will now fail without this change.

**Review by caohai — APPROVED** [review]
